### ISSUE TITLE
14 - Registrar la asistencia de alumno a clase

### ISSUE THREAD

**Issue by MatiasMillanao** [UNPSJB/secadmin#12]
Como empleado administrativo del SEC, hay que registrar la asistencia de un alumno a curso, el cual le permitirá al mismo, poder capacitarse y tener una posible salida laboral.

Datos:

- Dictado
- Clase
- Alumno

Criterios de aceptación:
-En caso de que el curso no exista, se vera un mensaje "No existe curso".
-En caso de que el curso exista, se buscará una clase de dicho curso.
-En caso de que la clase no exista, se verá un mensaje "No existe clase de curso"
-En caso de que la clase exista, se procederá a mostrar un listado con las clases.
-En caso de que la clase no se encuentre en el listado, se vera un mensaje "No existe la clase"
-En caso de que la clase se encuentre en el listado, se verifica fecha y hora de dicha clase.
-En caso de que la clase no se haya dictado en fecha y hora pactada, se vera un mensaje "No se puede verificar que la clase haya sido dictada en esa fecha y hora"
-En caso de que la clase se haya dictado en la fecha y hora pactada, se busca el listado con los alumnos de dicho curso.
-En caso de que no se encuentren alumnos en dicho curso, se verá un mensaje "No existen alumnos en el curso"
-En caso de que se encuentren alumnos en dicho curso, se procede a tomar el alumno requerido y registrar la asistencia del mismo al curso.
-En caso de querer seguir registrando asistencias de otros alumnos, se procede a repetir los pasos con los alumnos que se encuentren dentro del curso.

Tareas:

- [ ] Generar un pdf modelo de datos de asistencia
- [ ] Generar un endpoint que permita verificar que el curso buscado, existe.
- [ ] Generar un endpoint que permita verificar que la clase buscada, existe.
- [ ] Generar un endpoint que permita verificar que el curso buscado, se encuentre a través de fecha y hora.
- [ ] Generar un endpoint que permita verificar que en el curso buscado, se encuentren alumnos.
- [ ] Generar un endpoint que permita verificar que el alumno existe.

**Comment by MatiasMillanao** [comment]
- [ ] Generar la logica para que los alumnos encontrados, puedan tener una asistencia acorde a los dias en los cuales asistieron al curso.
- [ ] Generar la logica para que se pueda repetir el procedimiento las veces que fueran necesarias para registrar las asistencias de todos los alumnos a dicho curso.
- [ ] Generar la logica para que se muestre el listado de personas que asistieron al curso y la asistencia que tuvo cada uno en dicho curso.

**Comment by LucianoAlzugaray** [comment]
Podriamos dejar mas claro en la parte de "Datos" la relacion con clase?

**Comment by MatiasMillanao** [comment]
No entendí

**Comment by LucianoAlzugaray** [comment]
Dictado es una clase dentro de nuestro modelo de dominio que comprende un periodo, mientras que la clase es un dia o dias en particular en particular. Dentro de datos tenemos:

Datos:
-Nombre del curso
-Fecha de dictado
-Nombre y apellido de alumno


Fecha de dictado capaz es fecha de la clase. Sino capaz no la agarras vos y la agarra otro y busca meterle la fecha de dictado acá, cuando es la de la clase.

**Comment by MatiasMillanao** [comment]
claro, fecha de dictado seria el general, onda de 2 de junio a 20 de septiembre, y la fecha de clase son los dias y hora en los cuales seran las clases
